### PR TITLE
f64/f32: add MaxAbs and fused ConvolveValidMaxAbs[Multi] (peak detection)

### LINKED ISSUE #138: Add MaxAbs reduction and fused ConvolveValid+MaxAbs for f64/f32 (peak-detection hot path)
## Context

I built a pure-Go EBU R 128 / ITU-R BS.1770-4 loudness normalization library (audionorm) that uses this simd package on its hot paths. After profiling, the dominant remaining cost is true-peak estimation: a 4x polyphase oversampling FIR whose peak (max absolute value of the oversampled signal) I need.

I already moved the oversampling to `f64.ConvolveValid` (fed reversed kernels, since ConvolveValid is a correlation) and it now runs through the AVX dot-product path. Two primitives would let me remove the remaining scalar work and a scratch buffer entirely. Filing as requested so they can be designed/implemented in the library proper.

### 1. `MaxAbs` reduction (f64 and f32)

```go
// f64
func MaxAbs(a []float64) float64   // returns max(|a[i]|); 0 for an empty slice
// f32
func MaxAbs(a []float32) float32
```

Returns the maximum absolute value of a slice. This is the generic reduction; today I compute it with a scalar `for _, y := range out { if abs(y) > m { m = abs(y) } }` loop after each convolution, which shows up as ~7.5% of CPU in my profile and does not vectorize.

This is consistent with the existing integer API: `i8.MaxAbs(a []int8) int` and `i8.SumAbs` already exist, so `f64.MaxAbs`/`f32.MaxAbs` fill the obvious gap. Generally useful well beyond my use case: peak metering, clipping detection, normalization headroom, infinity-norm.

### 2. Fused valid-convolution + MaxAbs (f64, ideally f32)

```go
func ConvolveValidMaxAbs(signal, kernel []float64) float64
func ConvolveValidMaxAbsMulti(signal []float64, kernels [][]float64) float64
```

`ConvolveValidMaxAbs` returns `max(|valid-convolution output|)` without materializing the output slice. `ConvolveValidMaxAbsMulti` returns the single maximum across all kernels' outputs (one value).

This is the ideal oversampling/peak-detection primitive. The Multi form maps exactly onto polyphase true-peak: pass the N phase kernels, get back the peak of the reconstructed (oversampled) signal in one call. For me it would:
- remove the per-call output scratch buffer (`tpOut`) entirely (no need to store N convolution outputs),
- fuse the abs-max reduction into the convolution pass (no second scalar scan),
- collapse my per-channel inner logic (4 `ConvolveValid` calls + 4 abs-max scans + a buffer) into one call.

It is more specialized than `MaxAbs`, but peak-of-oversampled-signal is the canonical reason to run a polyphase FIR in audio, so it should be broadly applicable.

## API consistency

- Naming mirrors existing symbols: `ConvolveValid` / `ConvolveValidMulti` already exist; `ConvolveValidMaxAbs` / `ConvolveValidMaxAbsMulti` extend them. `MaxAbs` matches `i8.MaxAbs`.
- Pure-Go fallback expected on unsupported platforms, like the rest of the package.
- Empty-input behavior: suggest `MaxAbs` returns 0 for an empty slice (no panic).

## Priority

`MaxAbs` first (small, general, removes the scalar tail). `ConvolveValidMaxAbs[Multi]` is the bigger structural win for the true-peak path (removes scratch + fuses the reduction), but is more specialized.

## Lower priority (no new API)

The convolution currently dispatches to `dotProductAVX` (AVX2, 4-wide float64) on my machine. An AVX-512 (8-wide) path for `convolveValid64` / `dotProduct` on capable CPUs could roughly halve the dominant convolution cost. Noting it as an internal optimization, not part of this request.

## Notes

No rush on my end; audionorm has a working scalar/`ConvolveValid` path today. I will switch to these the moment they land (happy to keep a build-tagged fallback). Confirmed the integer analogue (`i8.MaxAbs`) already exists, so the float versions are a natural addition.

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds three primitives to both the `f64` and `f32` packages, as requested in #138 for a true-peak / peak-detection hot path:

- **`MaxAbs(a) T`** - the infinity norm `max_i |a[i]|`, 0 for an empty slice. A real vectorized reduction: AVX2 (`VANDPD`/`VANDPS` + `VMAXPD`/`VMAXPS`), SSE2/SSE (`ANDPD`/`ANDPS` + `MAXPD`/`MAXPS`), NEON (`FABS` + `FMAX`), with a pure-Go fallback. Mirrors the existing `Min`/`Max` kernels with the abs folded into each load. AVX-512 CPUs reuse the AVX2 kernel (no AVX-512 hardware here to validate a dedicated VL kernel; consistent with the existing variance/euclidean handling in `f32`).
- **`ConvolveValidMaxAbs(signal, kernel) T`** - `max(|valid-convolution output|)` without materializing the output slice. The abs-max is fused into the convolution pass over the already-dispatched SIMD `dotProduct`, so it drops both the per-output scratch store and the separate scalar abs-max scan a consumer writes today.
- **`ConvolveValidMaxAbsMulti(signal, kernels) T`** - the single peak across all kernels' outputs, the polyphase true-peak primitive. Mirrors `ConvolveValidMulti`'s equal-kernel-length contract (panics on mismatch).

Per the issue author's follow-up, the convolution speedup lever is the fusion (removing the output store + the ~9% scalar abs-max scan), not lane width, so `ConvolveValidMaxAbs[Multi]` is implemented as a Go-level fusion that keeps each per-output dot product on the SIMD path on every backend; no new convolution assembly. The standalone `MaxAbs` reduction is independently useful (peak metering, clipping detection, normalization headroom) and is the part that gets dedicated kernels.

Empty-input behavior: `MaxAbs` and both convolution helpers return 0 (no panic), as suggested in the issue.

## Related Issues

Fixes #138

## Test Plan

- [x] Go-reference parity + bit-exactness vs the per-element scalar path (`MaxAbs`)
- [x] Direct per-kernel parity tests so SSE/SSE2 is exercised on an AVX2 host
- [x] `ConvolveValidMaxAbs[Multi]` parity vs `ConvolveValid` + scalar abs-max (exact equality: both share the dispatched dot product)
- [x] Edge cases: empty/short signal, empty kernel, mismatched-kernel-length panic
- [x] `testing.AllocsPerRun == 0` for all three
- [x] Fuzz harnesses (SIMD vs Go reference, bit-exact)
- [x] Runnable examples
- [x] asmcheck validates every new NEON `WORD` against `golang.org/x/arch/arm64asm`
- [x] Full suite green on amd64 (SSE2/AVX2/Go) and on a Raspberry Pi 5 (NEON)
- [x] `golangci-lint run ./...` clean

## Summary by CodeRabbit

* **New Features**
  * Added `MaxAbs()` to compute the maximum absolute value of floating-point slices.
  * Added `ConvolveValidMaxAbs()` and `ConvolveValidMaxAbsMulti()` for efficient convolution true-peak detection without materializing intermediate outputs.
  * Available for both 32-bit and 64-bit floating-point types.
* **Documentation**
  * Extended operation tables, examples, and known limitations (including AMD64 small-slice fallback coverage).
* **Tests**
  * Added unit tests, fuzz coverage, and new Go examples to validate correctness and allocation-free behavior.

### CODE REVIEW

File: f64/convolve_maxabs_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+				kernel[i] = 0.5 - float64(i)*0.1
+			}
+			got := ConvolveValidMaxAbs(signal, kernel)
+			want := refConvolveValidMaxAbs(signal, kernel)
+			if math.Abs(got-want) > 1e-12*(1+math.Abs(want)) {
+				t.Errorf("sl=%d kl=%d: got %v want %v", sl, kl, got, want)
+			}
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f64/convolve_maxabs_test.go:41]: Fixed in d1cda19. Dropped the tolerance for exact equality, matching the f32 test. Both the fused path and ConvolveValid run the same dispatched dotProduct, so the peak is bit-identical.

File: f32/f32.go [hunk 2]
```diff
@@ -618,6 +634,72 @@ func ConvolveValidMulti(dsts [][]float32, signal []float32, kernels [][]float32)
...
+	}
+	kLen := len(kernels[0])
+	if kLen == 0 || len(signal) < kLen {
+		return 0
+	}
+	for i := 1; i < numKernels; i++ {
+		if len(kernels[i]) != kLen {
+			panic("simd: all kernels must have the same length")
+		}
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32.go:693]: Leaving this as is. The early return on `kLen == 0 || len(signal) < kLen` before the equal-length panic is intentional: it mirrors `ConvolveValidMulti` exactly (same order: empty/short signal returns before the mismatch check), and the doc says these helpers match that contract. Moving the panic ahead of the early return would make `ConvolveValidMaxAbsMulti` diverge from `ConvolveValidMulti`, which is the inconsistency I want to avoid. The polyphase use case always passes equal-length kernels, so the panic still fires for the realistic ragged-input case.
